### PR TITLE
Fix Legend storybook

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

Turns out that this wasn't a bug in Recharts, this was missing docs. I took the opportunity to fix the storybook.

## Related Issue

https://github.com/recharts/recharts/issues/4105

## Motivation and Context

Storybook controls are amazing!

## How Has This Been Tested?

npm run storybook

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a storybook story or extended an existing story to show my changes
- [x] All new and existing tests passed.

### CODE REVIEW

File: storybook/stories/API/props/Legend.ts [hunk 1]
```diff
@@ -75,53 +75,14 @@ export const LegendProps: StorybookArgs = {
...
-    defaultValue: 14,
+    defaultValue: 'line',
     table: {
       type: {
         summary: 'line | plainline | square | rect | circle | cross | diamond | star | triangle | wye',
       },
       category: 'General',
     },
   },
-  payload: {
```
Comment (ckifer) [inline storybook/stories/API/props/Legend.ts:86]: whats the intent of removing these? Just that they're internal?

File: storybook/stories/API/props/Legend.ts [hunk 2]
```diff
@@ -169,76 +130,4 @@ export const LegendProps: StorybookArgs = {
...
-  onClick: {
-    description: 'The customized event handler of click on the items in this group',
-    table: {
-      type: {
-        summary: 'Function',
-      },
-      category: 'General',
-    },
-  },
-  onMouseDown: {
```
Comment (ckifer) [inline storybook/stories/API/props/Legend.ts:181]: same here, why are these being removed?

**Review by ckifer — CHANGES_REQUESTED** [review]
Can we make the same changes we did to recharts.org to the props here?

Couple questions on the removals

**Comment by PavelVanecek** [comment]
Yeah I had to remove the props because they are set by element cloning and never meant to be set from outside. If they remain in the `LegendProps` then the storybook doesn't render.

I will update the doc too

**Comment by ckifer** [comment]
lol thats crazy misleading 😓. `Legend` has to have an `onClick` and some hover event at least though?... I know there are charts that use it

**Comment by ckifer** [comment]
Pretty sure all the events should still exist? https://recharts.org/en-US/api/Legend#onClick

**Comment by PavelVanecek** [comment]
I'll double check the events

**Comment by PavelVanecek** [comment]
So somehow, Legend still has onClick. And `payload`. And `dangerouslySetInnerHTML`. I have no clue where these are coming from: 
image

**Comment by ckifer** [comment]
they're coming from types. Storybook will use types to infer the args that are possible

**Review by ckifer — APPROVED** [review]
LGTM. Might need to figure out whats exactly going on with events eventually though

**Pull request merged**